x/ibc/02-client/keeper/client_test.go

```diff
@@ -25,32 +26,38 @@ func (suite *KeeperTestSuite) TestCreateClient() {
 
 	cases := []struct {
 		msg      string
-		params   params
+		clientID string
 		expPass  bool
 		expPanic bool
 	}{
-		{"success", params{testClientID, exported.Tendermint}, true, false},
-		{"client ID exists", params{testClientID, exported.Tendermint}, false, false},
-		{"client type exists", params{testClientID2, exported.Tendermint}, false, true},
-		{"invalid client type", params{testClientID3, invalidClientType}, false, false},
+		{"success", testClientID, true, false},
+		{"client ID exists", testClientID, false, false},
+		{"client type exists", testClientID2, false, true},
 	}
 
 	for i, tc := range cases {
 		tc := tc
 
 		if tc.expPanic {
 			suite.Require().Panics(func() {
-				suite.keeper.CreateClient(suite.ctx, tc.params.clientID, tc.params.clientType, suite.consensusState, trustingPeriod, ubdPeriod)
+				clientState, err := ibctmtypes.Initialize(tc.clientID, suite.consensusState, trustingPeriod, ubdPeriod)
+				suite.Require().NoError(err, "err on client state initialization")
+				suite.keeper.CreateClient(suite.ctx, clientState, suite.consensusState)
 			}, "Msg %d didn't panic: %s", i, tc.msg)
 		} else {
-			clientState, err := suite.keeper.CreateClient(suite.ctx, tc.params.clientID, tc.params.clientType, suite.consensusState, trustingPeriod, ubdPeriod)
+			clientState, err := ibctmtypes.Initialize(tc.clientID, suite.consensusState, trustingPeriod, ubdPeriod)
+			if tc.expPass {
```

Comment:
there are multiple `tc.expPass` that can be refactored

Reply:
Need to have multiple because the error can happen in 2 places now.

Want to make sure that if the error happened on the first step, the test failure message reflects this